src/components/modal/createModalLauncher.tsx

```diff
@@ -25,8 +25,8 @@ export const createRawModalLauncher =
   (componentProps?: P): ModalLauncher<D> =>
   (onModalClose) => {
     const { onClose, ...restModalProps } = modalProps;
-    const handleClose = (obj?: any) => {
-      onClose?.(null);
+    const handleClose = (event: any, obj?: any) => {
```

Comment:
Let's remove the `any` as it is getting us into trouble:
```suggestion
    const handleClose = (event?: KeyboardEvent | React.MouseEvent, obj?: any) => {
```

Also, above:
```
export type ComponentProps<D = unknown> = {
  onClose?: (event?: KeyboardEvent | React.MouseEvent, obj?: D) => void;
};
```